ortools/linear_solver/unittests/xpress_interface.cc

```diff
@@ -132,6 +142,24 @@ namespace operations_research {
       return (XPRSprob)solver_->underlying_solver();
     }
 
+    std::string getName(int n, int type) {
+      int namelength;
+      EXPECT_STATUS(XPRSgetintattrib(prob(), XPRS_NAMELENGTH, &namelength));
+
+      std::string name;
+      name.resize(8 * namelength + 1);
+      // looking for a single name i.e first = last; n=n
+      EXPECT_STATUS(XPRSgetnames(prob(), type, name.data(), n, n));
```

Comment:
`XPRSgetnames()` is deprecated since version 42.00 (Xpress 9.2). You should rather use [XPRSgetnamelist()](https://www.fico.com/fico-xpress-optimization/docs/latest/solver/optimizer/HTML/XPRSgetnamelist.html).
This has an API that can account for _any_ length of names and does not require to deal with whitespaces in names.

Reply:
Thanks Daniel !

Reply:
That looks like a good idea. However, `XPRSgetnamelist` and friends (`XPRS_NAMES_COLUMN`, `XPRS_NAMES_ROW`) are quite recent.

We should need to upgrade environment.h / environment.cc to include them.

We would make OR-Tools dependent on XPRESS 9.2+